doc/api/stream.md

```diff
@@ -90,60 +97,224 @@ server.listen(1337);
+##### Event: 'unpipe'
+
+* `src` {[Readable][] Stream} The source stream that
+  [unpiped][`stream.unpipe()`] this writable
+
+The `'unpipe'` event is emitted when the [`stream.unpipe()`][] method is called
+on a [Readable][] stream, removing this [Writable][] from its set of
+destinations.
+
+```js
+const writer = getWritableStreamSomehow();
+const reader = getReadableStreamSomehow();
+writer.on('unpipe', (src) => {
+  console.error('Something has stopped piping into the writer.');
+  assert.equal(src, reader);
+});
+reader.pipe(writer);
+reader.unpipe(writer);
+```
+
+##### writable.cork()
+
+The `writable.cork() method forces all written data to be buffered in memory.
+The buffered data will be flushed when either the [`stream.uncork()`][] or
+[`stream.end()`][stream-end] methods are called.
+
+##### writable.end([chunk][, encoding][, callback])
+
+* `chunk` {String|Buffer} Optional data to write
+* `encoding` {String} The encoding, if `chunk` is a String
+* `callback` {Function} Optional callback for when the stream is finished
+
+Calling the `writable.end()` method signals that no more data will be written
+to the [Writable][]. The optional `chunk` and `encoding` arguments allow one
+final additional chunk of data to be written immediately before closing the
+stream. If provided, the optional `callback` function is attached as a listener
+for the [`'finish'`][] event.
+
+Calling the [`stream.write()`][stream-write] method after calling
+[`stream.end()`][stream-end] will raise an error.
+
+```js
+// write 'hello, ' and then end with 'world!'
+const file = fs.createWriteStream('example.txt');
+file.write('hello, ');
+file.end('world!');
+// writing more now is not allowed!
+```
+
+##### writable.setDefaultEncoding(encoding)
+
+* `encoding` {String} The new default encoding
+* Return: `this`
+
+The `writable.setDefaultEncoding()` method sets the default `encoding` for a 
+[Writable][] stream.
+
+##### writable.uncork()
+
+The `writable.uncork()` method flushes all data buffered since
+[`stream.cork()`][] was called.
+
+##### writable.write(chunk[, encoding][, callback])
+
+* `chunk` {String|Buffer} The data to write
+* `encoding` {String} The encoding, if `chunk` is a String
+* `callback` {Function} Callback for when this chunk of data is flushed
+* Returns: {Boolean} `true` if the data was handled completely.
+
+The `writable.write()` method writes some data to the stream, and calls the
+supplied `callback` once the data has been fully handled. If an error
+occurs, the `callback` *may or may not* be called with the error as its
+first argument. To reliably detect write errors, add a listener for the
+`'error'` event.
+
+The return value indicates if writing should be continued or paused as the
+current time. If the written data had to be buffered internally, the return
+value will be `false` and further attempts to write data to the stream should
+be paused until the `'drain'` event is emitted. Otherwise, the return value
+will be `true`.
```

Comment:
This is incorrect — it will return `false` if the written chunk was buffered **and** the buffer exceeded `highWaterMark`. The intent here is that `highWaterMark` gives Writable streams an _optional_ cushion — we can only flush one write at a time, but we advertise that we're still available for writing _until_ our buffer is full during that flush. The reasoning behind this is that, for the resources Node is dealing with, it is expensive to make the roundtrip to turn them on or off (like turning a rusty garden hose spigot on or off), so we want to "downsample" backpressure in order to spend less time turning the emitting resource on-and-off-again.